### DISCUSSION TITLE
boto3 vs redshift_connector

### DISCUSSION THREAD

**Discussion by benhubsch** [aws/amazon-redshift-python-driver#195] [General] [answered]
We've been using boto3 and the RedshiftDataAPIService to fetch results from Redshift with relatively poor performance. After a long time spent trying to optimize our queries and batch them in different ways, today I just tried using this library instead, and the performance of is significantly better.

Cached queries which otherwise were taking up to 10 seconds to fetch via a combination of [`execute_statement`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/redshift-data/client/execute_statement.html) and [`get_statement_result`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/redshift-data/client/get_statement_result.html) take under 200 milliseconds.

Any insight into why this is the case?

**Comment by Brooke-white** [comment, marked as answer]
Hi @benhubsch , apologies for the late response I missed your thread. This is interesting to hear about the difference in performance you saw with boto3 + redshift-data vs redshift-connector. Unfortunately I am not familiar with the inner workings of boto3 and how it integrates with clients such as Redshift Data API. 

While I'm happy to hear redshift-connector is more performant for your workload, I'm sure the boto3 + Redshift Data API teams would be interested in hearing about your experience + details about the type of workload you're running to better understand the performance seen. I'd recommend to reach out to AWS Support who can help you get in touch with the boto3 + Data API teams.
